app/pages/menu.tsx

```diff
@@ -60,16 +60,31 @@ const Menu: BlitzPage<MenuProps> = ({ user }) => {
       </div>
       <Divider padded />
       <div>
-        {hasShop ? (
-          <>
-            <MenuListItem icon={<FiLayout />} title="My shop" />
-            <Link href={Routes.ManageProductsPage().pathname} passHref>
-              <MenuListItem as="a" icon={<FiGrid />} title="My products" />
-            </Link>
-            <Link href={Routes.TransactionHistory().pathname} passHref>
-              <MenuListItem as="a" icon={<FiList />} title="My balance" />
-            </Link>
-          </>
+        {shop ? (
+          shop.shopStatus === ShopStatus.APPROVED ? (
+            <>
+              <MenuListItem icon={<FiLayout />} title="My shop" />
+              <Link href={Routes.ManageProductsPage().pathname} passHref>
+                <MenuListItem as="a" icon={<FiGrid />} title="My products" />
+              </Link>
+              <Link href={Routes.TransactionHistory().pathname} passHref>
+                <MenuListItem as="a" icon={<FiList />} title="My balance" />
+              </Link>
+            </>
+          ) : shop.shopStatus === ShopStatus.DECLINED ? (
+            <>
+              <MenuListItem as="a" title="Your request has been declined" />
+              <Link href={Routes.RegisterPage().pathname} passHref>
+                <MenuListItem
+                  as="a"
+                  icon={<FiLayout />}
+                  title="Register new shop"
+                />
+              </Link>
+            </>
+          ) : shop.shopStatus === ShopStatus.REQUESTED ? (
+            <MenuListItem as="a" title="Your registration is in progress" />
```

Comment:
Maybe switch to div as it's not clickable

```suggestion
            <MenuListItem as="div" title="Your registration is in progress" />
```